src/main/java/edu/kit/datamanager/pit/elasticsearch/PidRecordElasticWrapper.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright 2023 Karlsruhe Institute of Technology.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package edu.kit.datamanager.pit.elasticsearch;
+
+import edu.kit.datamanager.pit.domain.Operations;
+import edu.kit.datamanager.pit.domain.PIDRecord;
+import edu.kit.datamanager.pit.pidsystem.impl.local.PidDatabaseObject;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import javax.persistence.ElementCollection;
+import javax.persistence.FetchType;
+
+import org.springframework.data.annotation.Id;
+import org.springframework.data.elasticsearch.annotations.DateFormat;
+import org.springframework.data.elasticsearch.annotations.Document;
+import org.springframework.data.elasticsearch.annotations.Field;
+import org.springframework.data.elasticsearch.annotations.FieldType;
+
+@Document(indexName = "typedpidmaker")
+public class PidRecordElasticWrapper {
+
+    @Id
+    private String pid;
+
+    @ElementCollection(fetch = FetchType.EAGER)
+    private Map<String, ArrayList<String>> attributes = new HashMap<>();
+
+    @Field(type = FieldType.Date, format = DateFormat.basic_date_time)
+    private Date created;
+
+    @Field(type = FieldType.Date, format = DateFormat.basic_date_time)
+    private Date lastUpdate;
+
+    @Field(type = FieldType.Text)
+    private List<String> read = new ArrayList<>();
+
+    public PidRecordElasticWrapper(PIDRecord pidRecord, Operations dateOperations) {
+        pid = pidRecord.getPid();
+        PidDatabaseObject simple = new PidDatabaseObject(pidRecord);
+        this.attributes = simple.getEntries();
+        this.read.add("anonymousUser");
+
+        try {
+            this.created = dateOperations.findDateCreated(pidRecord).orElse(null);
+            this.lastUpdate = dateOperations.findDateModified(pidRecord).orElse(null);
+        } catch (IOException e) {
+            e.printStackTrace();
```

Comment:
I would at least expect logging of the exception here, even if I cannot imaging any scenario where an exception may occur here.

Reply:
IO errors may occur when there is an issue resolving the types. Currently, I only use a dirty "see if the name matches one of our examples" in case there is no known type. Later I'd like to look for semantic relationships to known types, but this requires further work, so I skipped it as the feature was urgent. I am currently adding some documentation to the operations for this:

```java
/**
     * Tries to get the date when a FAIR DO was modified from a PID record.
     * 
     * Strategy:
     * - try to get it from known "dateModified" types
     * - as a fallback, try to get it by its human readable name
     * 
     * Semantic reasoning in some sense is planned but not yet supported.
     * 
     * @param pidRecord the record to extract the information from.
     * @return the date, if it could been extracted.
     * @throws IOException on IO errors regarding resolving types.
     */
```

Fix should be incoming shortly. Just have to figure out how to exclude the logger from being a field to be serialized.

Reply:
Logging is now included. It logs a message, the exception that has been thrown, and the PID for which the issue occurred.